Teensy/src/sdCard/sdCard.cpp

```diff
@@ -5,11 +5,135 @@
  * Detailed description, etc.
  */
 
+
 #include <Arduino.h>
+#include <SD.h>
+#include <SPI.h>
+#include <TimeLib.h>
+#include <DS1307RTC.h>
+
 #include "sdCard.h"
 
+File dataFile;
+String timeStamp;
+char * fileName;
+const int chipSelect = BUILTIN_SDCARD;
+
+
 extern bool sendDataToSdCard(sensorData_s *pSensorData)
 {
-    // Send data to SD card
+
+fileName = (char*) timeStamp.c_str();
+dataFile = SD.open(fileName, FILE_WRITE);
+
+  //if there was no problem opening the file => write to it
+  if (dataFile) {
+    Serial.print("writing to");
+    Serial.println(fileName);
+    
+    //Print the data to the file
+    dataFile.print(String(pSensorData->wheelSpeed_mph.fr) + ",");
+    dataFile.print(String(pSensorData->wheelSpeed_mph.fl) + ",");
+    dataFile.print(String(pSensorData->wheelSpeed_mph.rr) + ",");
+    dataFile.print(String(pSensorData->wheelSpeed_mph.rl) + ",");
+
+    dataFile.print(String(pSensorData->engineRev_rpm) + ",");
+
+    dataFile.print(String(pSensorData->damperPos_mm.fr) + ",");
+    dataFile.print(String(pSensorData->damperPos_mm.fl) + ",");
+    dataFile.print(String(pSensorData->damperPos_mm.rr) + ",");
+    dataFile.print(String(pSensorData->damperPos_mm.rl) + ",");
+
+    dataFile.print(String(pSensorData->gearPos) + ",");
+
+    dataFile.print(String(pSensorData->steeringWheelPos_degrees) + ",");
+
+    dataFile.print(String(pSensorData->gyro.x) + ",");
+    dataFile.print(String(pSensorData->gyro.y) + ",");
+    dataFile.print(String(pSensorData->gyro.z) + ",");
+
+    dataFile.print(String(pSensorData->batteryVoltage_dV) + ",");
+
+    dataFile.print(String(pSensorData->throttlePos_mm) + ",");
+
+    dataFile.print(String(pSensorData->fuelPressure_pa) + ",");
+
+    dataFile.println(String(pSensorData->time_ms) + ",");
+
+    dataFile.close();
+  } else {
+    //if the file couldn't be opend => print error message
+    Serial.print("error oppening the file :");
+    Serial.println(fileName); 
+  }
     return true;
-}
\ No newline at end of file
+}
+
+
+extern bool sdCard_init() 
+{
+
+  //Get the time to use for a timestamp
+  timeStamp = String("FS_DAQ_Log_" + String(year()) + "-" + String(month())\
```

Comment:
I'd prefer to work with a straight char array here instead of messing around with converting between strings and char arrays. 

snprintf is probably the best way to do this -
```
snprintf (gFilename, FILENAME_LEN, "FS_DAQ_Log_%s-%s-%s_%s:%s:%s.csv", year(), month(), day(), hour(), minute(), second());
```

Reply:
It wanted it to be referencing int dtype so I used "%d" instead of "%s", It also has a warning saying the "\0" isn't needed but I left it in as it is easy to change.

Reply:
Yep, I was thinking year() and that were returning strings but I can see now that you were converting them to strings before so they couldn't have been! Just looked it up and snprintf null terminates so you're all good to remove the `\0`